Upgrade: fix RKE2 is not upgraded in the single node case

bk201 opened the pull request:
**Problem:**
<!-- Explain the problem you are aiming to resolve in this PR. -->
RKE2 is pending for upgrade in single node upgrade case.

**Solution:**
<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->
The cluster is waiting for the `mcc-local-managed-system-upgrade-controller` bundle.  The bundle can't be ready due to no fleet-agent pod running. Previously, we scaled down fleet-agent to avoid hitting http:///github.com/rancher/fleet/issues/637. Since https://github.com/rancher/fleet/issues/637 is fixed, we can remove the workaround.

**Related Issue:**
https://github.com/harvester/harvester/issues/4119

**Test plan:**
<!-- Make sure tests pass on the Circle CI. -->
- Create a single node 1.1.2 cluster.
- Upgrade to the PR version; the upgrade should succeed.
**note**: need to test with https://github.com/harvester/harvester/pull/4149

bk201 committed:
Upgrade: fix RKE2 is not upgraded in the single node case

Do not scale down fleet-agent during single node upgrade.
Since https://github.com/rancher/fleet/issues/637 is fixed, we can remove
the workaround. This prevent the cluster from waiting for the
`mcc-local-managed-system-upgrade-controller` bundle.

Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

starbops reviewed (commented):
Script part LGTM, but we might also want to remove the corresponding part in the upgrade controller that scales fleet-agent back up, too.

starbops committed:
Remove the part that scales fleet-agent up

We remove the scale-down part for fleet-agent in the upgrade script,
the corresponding part in the upgrade controller that brings it back
should also be removed. This affects single node upgrade only.

Signed-off-by: Zespre Chang <zespre.chang@suse.com>

Vicente-Cheng reviewed (approved):
LGTM, tested with single node upgrade.

w13915984028 reviewed (approved):
LGTM, thanks.

guangbochen reviewed (approved):
LGTM, thanks.
